libnetwork/types/network.go

```diff
@@ -34,6 +34,10 @@ type ContainerNetwork interface {
 	// DefaultNetworkName will return the default network name
 	// for this interface.
 	DefaultNetworkName() string
+
+	// NetworkInfo return the network information about binary path,
+	// package version and program version.
+	NetworkInfo() map[string]interface{}
```

Comment:
This is super generic which makes it hard to use for clients as never never know what they are getting.
Also I don't like the current output, this just redundantly prints netavark and aardvark three times.

How about
```
type NetworkInfo struct {
	Backend string         `json:"backend"`
	Version string         `json:"version,omitempty"`
	Package string         `json:"package,omitempty"`
	Path    string         `json:"path,omitempty"`
	DNS     DNSNetworkInfo `json:"dns,omitempty"`
}

type DNSNetworkInfo struct {
	Version string `json:"version,omitempty"`
	Package string `json:"package,omitempty"`
	Path    string `json:"path,omitempty"`
}
```